codegen/core/src/main/scala/com/lightbend/akkasls/codegen/ModelBuilder.scala

```diff
@@ -484,11 +488,32 @@ object ModelBuilder {
     if (ScalarType.isScalarType(resolvedName))
       ScalarTypeArgument(ScalarType(resolvedName))
     else {
-      val descriptor = lookupDescriptor(revolvedPackage, resolvedName, additionalDescriptors)
-      MessageTypeArgument(resolveFullyQualifiedMessageType(resolvedName, descriptor, additionalDescriptors))
+      lookupDomainDescriptor(revolvedPackage, resolvedName, additionalDescriptors)
+        .map { descriptor =>
+          MessageTypeArgument(resolveFullyQualifiedMessageType(resolvedName, descriptor, additionalDescriptors))
+        }
+        .getOrElse {
+          MessageTypeArgument(buildFullyQualifiedNameForPojos(revolvedPackage, resolvedName))
+        }
```

Comment:
So if I read this right, if a user misspells the stringly typed state annotation so it does not match a message, it will silently continue and expect it to be a POJO, wouldn't it be better with something more explicit in the annotations so that we can know the intent of the user and fail accordingly instead?

Reply:
Yes, I think so. We can add something, but I prefer to add it later. I don't want now to go the full cycle of changing annotations in the framework. 

For the moment, this PR is more important to get the POJO basics out. We can then fine tune it on other iterations.

Reply:
Could be done without any changes to the annotations, with some in-string prefix perhaps? `state: "java:MyState"` or `state: "pojo:MyState"`. That way it would also nicely fail if you tried to just move it to some other SDK/codegen.

Reply:
hmm, that's sound interesting

I will experiment with it, but in another PR. This in one is getting too big.